rsocket-core/src/main/java/io/rsocket/buffer/AbstractReadOnlyReferenceCountedByteBuf.java

```diff
@@ -0,0 +1,176 @@
+package io.rsocket.buffer;
+
+import io.netty.buffer.AbstractReferenceCountedByteBuf;
+import io.netty.buffer.ByteBuf;
+import java.io.InputStream;
+import java.nio.ByteBuffer;
+import java.nio.ReadOnlyBufferException;
+import java.nio.channels.FileChannel;
+import java.nio.channels.ScatteringByteChannel;
+
+abstract class AbstractReadOnlyReferenceCountedByteBuf extends AbstractReferenceCountedByteBuf {
```

Comment:
The netty docs deprecated similar classes and suggest using Unpooled.wrappedUnmodifiableBuffer.  Also the docs call out slice as an issue, how is this handled?

https://netty.io/4.1/api/io/netty/buffer/Unpooled.html#wrappedUnmodifiableBuffer-io.netty.buffer.ByteBuf...-

Reply:
It is deprecated but there are no other options to extend. TupledByteBuf designed to be readOnly and I don't wanna create few more objects to runtime just in order to use what netty gives. therefore I mimic the same what readonly bytebuf does